### PR TITLE
lavfi/{denoise,procamp,scale,sharpness}_vaapi: add passthrough mode

### PR DESCRIPTION
**Pull request by feiwan1**

### CODE REVIEW

**Comment by uartie** [comment]
This change introduces test failures in VPP contrast and saturation when value set at 0.5 and causes a passthrough result.

**Comment by uartie** [comment]
> This change introduces test failures in VPP contrast and saturation when value set at 0.5 and causes a passthrough result.

```
$ ffmpeg ... 'format=nv12|vaapi,hwupload,procamp_vaapi=s=0.5,hwdownload,format=nv12' ...
<snip>
[procamp_vaapi @ 0x55d66f19d100] Using VAAPI filter passthrough mode.
<snip>
psnr:expect = [100.0, 37.5973, 43.2666, 100.0, 40.1115, 44.8443]
psnr:actual = [100.0, 100.0, 100.0, 100.0, 100.0, 100.0]
```

File: libavfilter/vf_procamp_vaapi.c [hunk 1]
```diff
@@ -179,11 +182,18 @@ static int procamp_vaapi_filter_frame(AVFilterLink *inlink, AVFrame *input_frame
...
 
     ff_vaapi_vpp_ctx_init(avctx);
     vpp_ctx->pipeline_uninit     = ff_vaapi_vpp_pipeline_uninit;
     vpp_ctx->build_filter_params = procamp_vaapi_build_filter_params;
     vpp_ctx->output_format       = AV_PIX_FMT_NONE;
+    if (abs(ctx->saturation - SATURATION_DEFAULT) < eps &&
+        abs(ctx->bright - BRIGHTNESS_DEFAULT) < eps &&
+        abs(ctx->contrast - CONTRAST_DEFAULT) < eps &&
+        abs(ctx->hue - HUE_DEFAULT) < eps)
+        vpp_ctx->passthrough = 1;
```
Comment (uartie) [inline libavfilter/vf_procamp_vaapi.c]: Use `fabs` to fix test regressions

**Review by uartie — CHANGES_REQUESTED** [review]

File: libavfilter/vf_procamp_vaapi.c (hunk 1 above)
Comment (feiwan1) [reply libavfilter/vf_procamp_vaapi.c]: Fixed. Thanks.

**Comment by uartie** [comment]
LGTM

**Comment by uartie** [comment]
Maybe change `{denoise,procamp,scale,sharpness}_vaapi` to `vaapi_vpp` in commit message title.

**Comment by feiwan1** [comment]
> Maybe change `{denoise,procamp,scale,sharpness}_vaapi` to `vaapi_vpp` in commit message title.

vaapi_vpp gives more tendency to summary the changes in base class of vaapi_vpp.c, and better to seperate and commit other vaapi filter changes into single ones like:
1. vaapi_vpp: base function change.
2. scale_vaapi: passthrough mode.
3. ...

So I perfer using current way with {} to stress changes to the mentioned vaapi filters. And the style is widely used in ffmpeg like:

commit c889248647780753ec8b05138c7de4a707adb106
Refs: n5.2-dev-1333-gc889248647
Author:     Jan Ekström <jeebjp@gmail.com>
AuthorDate: Wed Oct 19 11:44:03 2022 +0200
Commit:     Thilo Borgmann <thilo.borgmann@mail.de>
CommitDate: Wed Oct 19 11:53:52 2022 +0200

    ffmpeg: Add display_{rotation, hflip, vflip} options

**Comment by feiwan1** [comment]
patch submitted to  https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=9025. Close.

**Pull request closed without merging**